### DISCUSSION TITLE
Updating

### DISCUSSION THREAD

**Discussion by ShoveD01** [bangcorrupt/hacktribe#264] [General]
I have had hacktribe installed since release, as I've not updated, is the updating procedure just the normal firmware update process? 
Has there been much added along the way?

**Comment by bangcorrupt** [comment]
Yes, apply the latest patch to sampler firmware version 2.02 every time.  For older versions, put the update file under `KORG/electribe sampler/System`.  After updating, copy your `electribe sampler` directory and rename it to `hacktribe`.

There have been a lot of bug fixes and performance improvements.  There is also an expanded MIDI implementation and some new delay IFX.   The [`experimental`](https://github.com/bangcorrupt/hacktribe/tree/experimental) branch has MIDI soft thru and sends motion sequences as CC, but it can be unstable when sending large amounts of MIDI data.

**Comment by ShoveD01** [comment]
Thanks, sounds good .What I was meaning was, as I already have hacktribe installed can I just update the downloaded file, or do I have to patch things first, as though I was installing from scratch.

**Reply by bangcorrupt** [reply]
You need to patch the factory firmware, you don't need to edit the file header.

**Comment by ShoveD01** [comment]
Does the event recorder now send midi data?

**Reply by bangcorrupt** [reply]
No, I haven't done anything with event recording yet.

**Comment by ShoveD01** [comment]
Just noticed what must be a bug. The filter cutoff knob is sending a data increment message out along with its usual cc, it certainly didn’t with my last but probably original version.  Nice work by the way.

**Reply by bangcorrupt** [reply]
It's a feature, all of the controls now send [NRPN](https://github.com/bangcorrupt/hacktribe/wiki/MIDI#nrpn) messages.  In a future update I'll add options to filter and turn it off.

**Comment by ShoveD01** [comment]
From what I’ve monitored it’s only certain controls e.g the resonance  only sends its regular cc.

**Comment by ShoveD01** [comment]
As a matter of interest, as a feature, why does it do this, it also appears to be recording the nrpn messages as motion data too.

**Comment by bangcorrupt** [comment]
Ok, looks like the knobs and pads don't send NRPN, because they already send CC and notes.  The filter encoder and some of the buttons send NRPN and CC because it was easier to program this way.

>  recording the nrpn messages as motion data

If you installed from the experimental branch motion sequences will be sent as MIDI, it's buggy.

**Comment by ShoveD01** [comment]
I asked because usually with nrpn messages the accompanied data message is I believe cc6 , the inc/dec would only be recognised by an encoder probably internally, I can’t think of any device that would respond to that message.Also, is there yet a method of importing/exporting ifx and grooves?

**Comment by ShoveD01** [comment]
Just made an fx parameter editor in lemur on the iPad, it really adds a massive amount to this machine. Is it only the ifx on slot 1 that have its Params edited, or can the mfx be edited too?

**Comment by bangcorrupt** [comment]
> method of importing/exporting ifx and grooves

The [`ht-cli`](https://github.com/bangcorrupt/hacktribe-editor-legacy/tree/ht-cli) branch of Hacktribe Editor has support for [FX editing](https://github.com/bangcorrupt/hacktribe-editor-legacy/wiki/FX-Preset-Editing), there is an old [script](https://github.com/bangcorrupt/hacktribe/blob/main/scripts/e2_groove_template.py) demonstrating groove editing.  Neither can be saved to the flash yet, so we have to save to the computer and load them every time.  

>  Is it only the ifx on slot 1 that have its Params edited, or can the mfx be edited too?

MFX are on slot 0x20.  Each IFX channel has 2 slots (most only use 1).  If you're sending NRPN directly, use NRPN-LSB to set FX slot from 0 to 0x20.  If you start `ht-cli`, then use the `run` command it will [map CC and note messages to NRPN](https://github.com/bangcorrupt/hacktribe-editor-legacy/wiki/MIDI-Control).
